tests/ui/offset-of/offset-of-dst-field.rs

```diff
@@ -26,8 +26,28 @@ struct Gamma {
     z: Extern,
 }
 
+struct Delta<T: ?Sized> {
+    x: u8,
+    y: u16,
+    z: T,
+}
+
 fn main() {
     offset_of!(Alpha, z); //~ ERROR the size for values of type
     offset_of!(Beta, z); //~ ERROR the size for values of type
     offset_of!(Gamma, z); //~ ERROR the size for values of type
 }
+
+fn delta() {
+    offset_of!(Delta<u8>, z); // compiles fine
```

Comment:
Maybe we should move fine-compiling parts to separate files, to ensure they really compile? (besides, I think there already are test that have the same compiling code, so maybe this and `generic_with_sized_type` are just redundant)